Add missing entities in template inheritance.

glafarge opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adds missing page to template inheritance, adds missing parameters to existing controllers and fix some typos
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #18107.
| How to test?  | Even if special templates  `catalog/listing/best-sales.tpl`,  `catalog/listing/new-products.tpl`, `catalog/listing/prices-drop.tpl`, `catalog/listing/search.tpl` are missing, the shop must display the `catalog/listing/product-list.tpl` as a default template.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/18125)
<!-- Reviewable:end -->

glafarge committed:
Add missing entities in template inheritance.

glafarge committed:
FIX PHP CS

PierreRambaud commented on classes/Smarty/TemplateFinder.php:
```diff
@@ -35,7 +35,7 @@ class TemplateFinderCore
     private $extension;
     private $productListEntities = ['category', 'manufacturer', 'supplier'];
-    private $productListSearchEntities = ['search', 'price-drop', 'best-sale'];
+    private $productListSearchEntities = ['search', 'prices-drop', 'best-sales', 'new-products'];
```
Changing these strings is a BC :thinking: 
What if someone already used `price-drop` or `best-sale`?
Should it be?
```suggestion
    private $productListSearchEntities = ['search', 'price-drop', 'best-sale', 'prices-drop', 'best-sales', 'new-products'];
```

glafarge replied:
This is how it's named everywhere (plural).
But your suggestion is a nice improvement ! It'll be more tolerant and avoid errors 👌

glafarge committed:
Update classes/Smarty/TemplateFinder.php

Add tolerance on plural/singular forms of templates filenames in template inheritance

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

khouloudbelguith commented:
Hi @glafarge,

It is ok for `best-sales,` `new-products,` and `prices-drop` pages :heavy_check_mark: 

Thanks!

Progi1984 commented:
Thanks @glafarge
